src/lib.rs

```diff
@@ -732,36 +732,119 @@ impl<'a, R: Rng> Iterator for AsciiGenerator<'a, R> {
-    /// let seed: &[_] = &[1, 2, 3, 4];
-    /// let mut rng: StdRng = SeedableRng::from_seed(seed);
-    /// println!("{}", rng.gen::<f64>());
-    /// rng.reseed(&[5, 6, 7, 8]);
-    /// println!("{}", rng.gen::<f64>());
-    /// ```
-    fn reseed(&mut self, Seed);
+    /// PRNG implementations are allowed to assume that bits in the seed are
+    /// well distributed. That means usually that the number of one and zero
+    /// bits are about equal, and values like 0, 1 and (size - 1) are unlikely.
+    ///
+    /// PRNG implementations are recommended to be reproducible. A PRNG seeded
+    /// using this function with a fixed seed should produce the same sequence
+    /// of output in the future and on different architectures (with for example
+    /// different endianness).
+    ///
+    /// It is however not required that this function yield the same state as a
+    /// reference implementation of the PRNG given equivalent seed; if necessary
+    /// another constructor can be used.
+    fn from_seed(seed: Self::Seed) -> Self;
 
-    /// Create a new RNG with the given seed.
+    /// Create a new PRNG seeded from another `Rng`.
     ///
-    /// # Example
+    /// This is the recommended way to initialize PRNGs. See the `NewRng`
+    /// trait that provides a convenient `new` method using `from_rng` and a
+    /// good entropy source.
     ///
-    /// ```rust
-    /// use rand::{Rng, SeedableRng, StdRng};
+    /// It is recommended to use a good source of randomness to initialize the
+    /// PRNG. Otherwise small PRNG's could show statistical bias in the first
+    /// couple of results, and possibly not use their entire period well.
+    /// Cryptographic PRNG's can be less secure or even insecure when they are
+    /// seeded from a non-cryptographic PRNG.
     ///
-    /// let seed: &[_] = &[1, 2, 3, 4];
-    /// let mut rng: StdRng = SeedableRng::from_seed(seed);
-    /// println!("{}", rng.gen::<f64>());
-    /// ```
-    fn from_seed(seed: Seed) -> Self;
+    /// Examples of good RNG's for seeding are entropy sources like `OsRng` and
+    /// `JitterRng`. Also cryptographically secure PRNG's (like `thread_rng`)
+    /// can be used without hesitation.
+    ///
+    /// Seeding a small PRNG from another small PRNG is be possible, but
+    /// something to be careful with. An extreme example of how this can go
+    /// wrong is seeding an Xorshift RNG from another Xorshift RNG. That will
+    /// effectively clone the generator.
+    ///
+    /// PRNG implementations are allowed to assume that a good RNG is provided
+    /// for seeding, and that it is cryptographically secure when appropriate.
+    /// There are no reproducibility requirements like endianness conversion.
+    fn from_rng<R: Rng>(mut rng: R) -> Result<Self, Error> {
+        let mut seed = Self::Seed::default();
+        rng.try_fill_bytes(seed.as_mut())?;
+        Ok(Self::from_seed(seed))
+    }
+}
+
+
+/// Seeding mechanism for PRNGs, providing a `new` function.
```

Comment:
A convenient way to seed new algorithmic generators, otherwise known as pseudo-random number generators (PRNGs).

(Replace with PRNG on next line)